ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,108 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
             sub_faces.extend(face.sub_faces_by_ratio(ratio))
         return sub_faces
 
+    def _sub_faces_by_dimensions(self, base_plane, rect_base, rect_height,
+                                 sub_face_height, sub_face_width,
+                                 sill_height, dist_breakup):
+        """Get a list of rectangular Face3D objects using parameters to define them.
+
+        Args:
+            rect_base: A number indicating the length of the base of the
+                parent rectangle.
+            rect_height: A number indicating the length of the height of the
+                parent rectangle.
+            sub_face_height: A number for the target height of the output rectangles.
+                If the ratio is too large for the height, the ratio will take 
+                precedence and the sub-rectangle height will be larger than this value.
+            sub_face_width: A number for the target width of the output rectangles.
+                If the ratio is too large for the width, the ratio will take 
+                precedence and the sub-rectangle width will be larger than this value.
+            dist_breakup: A number for the target breakup separation distance between
+                individual sub-rectangle centerlines.  If this number is larger than
+                the parent rectangle base, only one sub-rectangle will be produced.
+            sill_height: A number for the target height above the bottom edge of
+                the rectangle to start the sub-rectangles. If the ratio is too
+                large for the height, the ratio will take precedence and the
+                sub-rectangle height will be smaller than this value.
+            base_plane: A Plane object in which the rectangle exists.
+                The origin of this plane will be the lower left corner of the
+                rectangle and the X and Y axes will form the sides.
+                Default is the world XY plane.
+
+        Returns:
+            A list of Face3D objects for sub faces.
+        """
+        # if sub_face_height > rect_height, set it to just under rect_height
+        sub_face_height_final = (
+            rect_height - 0.01 if sub_face_height > rect_height - 0.01
+            else sub_face_height)
```

Comment:
I appreciate you trying to respect the 89 character limit but you don't need parentheses here and a better way to write it would be with `\` at the end of lines that are continued on the next one. [Here's an example](https://github.com/ladybug-tools/ladybug-geometry/blob/master/ladybug_geometry/geometry3d/face.py#L1366-L1367). Also, I think that we should make this like so:
```
rect_height - 0.01 * rect_height if sub_face_height > rect_height else sub_face_height
```
This method will get used in all sorts of units systems and an absolute offset of 0.01 might not be practical if the units are millimeters, for example.